File: problem_solving_algo_and_ds/python/data_structures/base_converter.py
from stack import *

def base_converter(num, base):
    stack = Stack()
    digits = "0123456789ABCDEF"

    while num > 0:
        rem = num % base
        stack.push(rem)
        num = num // base

    conv_num = ""

    while not stack.isEmpty():
        conv_num += digits[stack.pop()]

    return conv_num

print(base_converter(34, 8))

File: problem_solving_algo_and_ds/ruby/data_structures/base_converter.rb
require_relative 'stack'

def base_converter(num, base)
  stack = Stack.new
  digits = "0123456789ABCDEF"

  while num > 0
    rem = num % base
    stack.push(rem)
    num = num / base
  end

  conv_num = ""

  while !stack.is_empty
    conv_num += digits[stack.pop]
  end

  return conv_num

end

p base_converter(34, 8)
